index.bs

```diff
@@ -560,79 +560,246 @@ urlPrefix: https://w3c.github.io/FileAPI/#dfn-; type: dfn;
+
+				1. If |representation|'s [=representation/MIME type=] is |mimeType|, then:
+
+					1. Let |p1| be the |representation|'s [=representation/data=].
+
+						1. [=promise/React=] to |p1|:
+
+							1. If |p1| was fulfilled with value |v|, then:
+
+								1. If |v| is a {{DOMString}}, then follow the below steps:
+
+									1. Let |v1| be the result of [=/encoding=] |v|.
+
+									1. Let |blobData| be a [=Blob=] created using |v1| with its {{Blob/type}} set to |mimeType|, [=serialize a MIME type|serialized=].
+
+									1. Resolve |p| with |blobData|.
+
+									1. Return |p|.
 
-		enum PresentationStyle { "unspecified", "inline", "attachment" };
+								1. If |v| is a [=Blob=], then follow the below steps:
 
-		dictionary ClipboardItemOptions {
-		  PresentationStyle presentationStyle = "unspecified";
+									1. Resolve |p| with |v|.
+
+									1. Return |p|.
+
+							1. If |p1| was rejected, then:
+
+								1. [=Reject=] |p| with {{"NotFoundError"}} DOMException in |realm|.
+
+								1. Return |p|.
+
+			1. [=Reject=] |p| with {{"NotFoundError"}} DOMException in |realm|.
+
+			1. Return |p|.
+
+	<h3 id="clipboard-interface">Clipboard Interface</h3>
+
+		<pre class="idl" data-highlight="webidl">
+		typedef sequence&lt;ClipboardItem&gt; ClipboardItems;
+
+		[SecureContext, Exposed=Window]
+		interface Clipboard : EventTarget {
+			Promise&lt;ClipboardItems&gt; read();
+			Promise&lt;DOMString&gt; readText();
+			Promise&lt;undefined&gt; write(ClipboardItems data);
+			Promise&lt;undefined&gt; writeText(DOMString data);
 		};
 		</pre>
 
-		<div id="clipboard-idl" dfn-for="Clipboard">
+		A <dfn>Clipboard</dfn> may contain <dfn>Clipboard Items</dfn> which has one or more [=clipboard item=]s. Multiple [=clipboard item=]s are only supported on platforms such as iOS/iPadOS.
+		A web author needs to create a |data| which is a [=Clipboard items=] object in order to write content to [=system clipboard=] using the {{Clipboard/write(data)}} method.
+		{{Clipboard/read()}} returns a [=Promise=] to [=Clipboard items=] object that represents contents of [=system clipboard=].
+		A [=clipboard item=] can be read from [=clipboard items=].
+
+		The <dfn>clipboard task source</dfn> is triggered in response to reading or writing of [=system clipboard data=].
 
+		<div id="clipboard-idl" dfn-for="Clipboard">
 		<div class="algorithm" data-algorithm="clipboard-read">
 		<h4 method for="Clipboard">read()</h4>
 		The {{Clipboard/read()}} method must run these steps:
 
-			1. Let |p| be a new [=Promise=].
+			1. Let |realm| be [=this=]'s [=relevant realm=].
+
+			1. Let |p| be [=a new promise=] in |realm|.
 
 			1. Run the following steps [=in parallel=]:
 
-				1. Let |r| be the result of running [=check clipboard read permission=] [=in parallel=]
+				1. Let |r| be the result of running [=check clipboard read permission=].
+
+					Note: Clipboard permission is not supported on Safari. However, the read() method must be called inside
```

Comment:
Since this is a UA specific quirk and to limit the scope of this PR, I propose to leave it as is. If we decide to change anything in the future, we should bring this up in EditingWG meetings and get consensus from all browser vendors.